Facebook Custom Audiences (Actions) documentation (#6913)
segmentio/segment-docs: src/connections/destinations/catalog/actions-facebook-custom-audiences/index.md

This destination sends audiences, or lists of users, from your data warehouse to Facebook Custom Audiences using Reverse ETL. Syncs can add or remove users from your Facebook Custom Audiences as records are created, updated, or deleted in your warehouse. 

> info "This destination only supports **Reverse ETL sources**
> This destination only supports [Reverse ETL sources](/docs/connections/reverse-etl/#what-warehouse-data-sources-does-segment-support). To connect Facebook Custom Audiences to Twilio Engage, use the [Facebook Custom Audiences](/docs/connections/destinations/catalog/personas-facebook-custom-audiences/) destination. 

## Getting started

### Prerequisites
- A Reverse ETL source already set up. If you don't yet have a Reverse ETL source, follow the instructions in Segment's [Reverse ETL documentation](/docs/connections/reverse-etl/#getting-started). Segment recommends setting your External ID as your primary key for your Reverse ETL model, as you'll need an External ID to remove users from your custom audiences. 

> - A Facebook Custom Audiences account with the following permissions: 

Author: @marinhero @mgutten - do you all know which permissions (if any) someone would need?

Reviewer: Let's update this to:

- A Facebook account with the following permissions for the target Facebook Ads Account(s):
- [ads_management](https://developers.facebook.com/docs/permissions#ads_management)

---
title: Facebook Custom Audiences (Actions)
id: 645d5fc12eb891cf0a93fe4b

> hidden: true

Author: if this is going into public beta we should remove this line

```suggestion
```

<ol style="counter-reset: none;">
  <li value="6" markdown=1>
  Set how often your model syncs by setting the [Sync schedule](/docs/connections/reverse-etl/#step-4-create-mappings). 
  </li>
  <li value="7" markdown=1>
  Select or create an audience in Facebook to sync your data with. Click the **Select or create audience in Facebook** button to save the audience ID to your mapping. 
  </li>
  <li value="8" markdown=1>
  Map your model columns to the appropriate Facebook Custom Audience parameters. For more context about data formatting, see the [Sync Audience](#sync-audience) and [Data processing](#data-processing) documentation. 
  <ul> 
    <li> Map External ID to a unique user identifier from your system (like User ID, CRM ID, or anonymous ID.) Segment recommends using the External ID column as your primary key when setting up your Reverse ETL model so you can more easily remove users from your custom audience. External ID is the only field Facebook requires. </li> 
    <li> Segment recommends mapping as many parameters as you have available in your source model so that you can increase your match rates. </li> 
  </ul>
  </li>
  <li value ="9" markdown=1>

>   Send a test record. If successful, you should see a 200 response in Segment and one added record to your custom audience. To verify that the record was successfully added to your custom audience, open Facebook Ads Manager and navigate to **Audiences > {Audience Name} > History**.

Reviewer 2: ```suggestion
  Send a test record. If successful, you should see a `200` response in Segment and one added record to your custom audience. To verify that the record was successfully added to your custom audience, open Facebook Ads Manager and navigate to **Audiences > {Audience Name} > History**.
```

---
title: Facebook Custom Audiences (Actions)
id: 645d5fc12eb891cf0a93fe4b

> beta: true

Author: @mgutten This is a feature, not a bug! For destinations in Private Beta, this box doesn't render. When the destination is moved into Public Beta tomorrow and the docs catalog updates, the box will render correctly!

strat: facebook
hide_action:
  - id: 3JEx23g4qgFzSECyYopNA4
    name: "Add"
  - id: 99Dj24PD8pAweMvTnM8vD8
    name: "Sync Reverse ETL"
  - id: eG6ydbfKbfJvQZ8gQpgakz
    name: "Sync Engage"
  - id: p74MiTEFmUUeoC7SKcT7Ri
    name: "Remove"
---

Deliver effective Facebook Ads campaigns by defining and syncing [custom audiences](https://developers.facebook.com/docs/marketing-api/audiences/guides/custom-audiences){:target="_blank”} to power ad suppression, acquisition (using lookalikes), retargeting, and more.

This destination sends audiences, or lists of users, from your data warehouse to Facebook Custom Audiences using Reverse ETL. Syncs can add or remove users from your Facebook Custom Audiences as records are created, updated, or deleted in your warehouse. 

> info "This destination only supports **Reverse ETL sources**
> This destination only supports [Reverse ETL sources](/docs/connections/reverse-etl/#what-warehouse-data-sources-does-segment-support). To connect Facebook Custom Audiences to Twilio Engage, use the [Facebook Custom Audiences](/docs/connections/destinations/catalog/personas-facebook-custom-audiences/) destination. 

## Getting started

### Prerequisites

> - A Reverse ETL source already set up. If you don't yet have a Reverse ETL source, follow the instructions in Segment's [Reverse ETL documentation](/docs/connections/reverse-etl/#getting-started). Segment recommends setting your External ID as your primary key for your Reverse ETL model, as you'll need an External ID to remove users from your custom audiences. 

Author: ```suggestion
- A Reverse ETL source already set up. If you don't yet have a Reverse ETL source, follow the instructions in Segment's [Reverse ETL documentation](/docs/connections/reverse-etl/#getting-started). Segment recommends setting your [External ID](#available-actions) as your primary key for your Reverse ETL model, as you'll need an External ID to remove users from your custom audiences. 
```

> - A Reverse ETL source already set up. If you don't yet have a Reverse ETL source, follow the instructions in Segment's [Reverse ETL documentation](/docs/connections/reverse-etl/#getting-started). Segment recommends setting your External ID as your primary key for your Reverse ETL model, as you'll need an External ID to remove users from your custom audiences. 

Author: ```suggestion
- A Reverse ETL source already set up. If you don't yet have a Reverse ETL source, follow the instructions in Segment's [Reverse ETL documentation](/docs/connections/reverse-etl/#getting-started). Segment recommends setting your [External ID](/docs/connections/destinations/catalog/actions-facebook-conversions-api/#destination-settings) as the primary key for your Reverse ETL model, as you'll need an External ID to remove users from your custom audiences. 
```

> - A Facebook Custom Audiences account with the following permissions: 
>   - 

Author: ```suggestion
- A Facebook account with [ads_management](https://developers.facebook.com/docs/permissions#ads_management){:target="_blank”} permissions for the target Facebook Ads Account(s).
```

4. Enter a descriptive name for your mapping. Segment recommends a name that includes both the audience name and sync mode, for example, `Loyalty Users (Add)`. 
5. Under **Select record to map and send**, select **Added or updated records**. The Added or updated records sync mode both adds new records and attempts to re-add any updated records to the custom audience. Adding updated records to your destination enables better match rates, as more user identifiers are added to the source model over time.

> warning "Added or updated records is the only supported additive sync mode"
> Selecting any other sync mode might lead to sync failures with the Facebook Custom Audiences (Actions) destination. 

<ol style="counter-reset: none;">
  <li value="6" markdown=1>
  Set how often your model syncs by setting the [Sync schedule](/docs/connections/reverse-etl/#step-4-create-mappings). 
  </li>
  <li value="7" markdown=1>
  Select or create an audience in Facebook to sync your data with. Click the **Select or create audience in Facebook** button to save the audience ID to your mapping. 
  </li>
  <li value="8" markdown=1>

>   Map your model columns to the appropriate Facebook Custom Audience parameters. For more context about data formatting, see the [Sync audience](#sync-audience) and [Data processing](#data-processing) documentation. 

Author: ```suggestion
  Map your model columns to the appropriate Facebook Custom Audience parameters. For more context about data formatting, see the [Sync Audience](#sync-audience) and [Data processing](#data-processing) documentation. 
```

  <ul> 

>     <li> Map External ID to a unique user identifier from your system (like User ID, CRM ID, or anonymous ID.) Segment recommends using the External ID column as your primary key when setting up your Reverse ETL model so you can more easily remove users from your custom audience. External ID is the only field Segment requires you to map. </li> 

Reviewer: "...only field Segment requires you to map..." -> "...only field Facebook requires."

>     <li> Map External ID to a unique user identifier from your system (like User ID, CRM ID, or anonymous ID.) Segment recommends using the External ID column as your primary key when setting up your Reverse ETL model so you can more easily remove users from your custom audience. External ID is the only field Segment requires you to map. </li> 

Author: ```suggestion
    <li> Map External ID to a unique user identifier from your system (like User ID, CRM ID, or anonymous ID.) Segment recommends using the External ID column as your primary key when setting up your Reverse ETL model so you can more easily remove users from your custom audience. External ID is the only field Facebook requires. </li> 
```

  </li>
  <li value="10" markdown=1>
  Click **Save Mapping** and enable the mapping. 
  </li>
</ol>

### Remove users from a Custom Audience

1. Navigate to **Connections > Sources** and select your Reverse ETL source.
2. On the Models page, select the model you'd like to use and click **Add Mapping**. 
3. Select the Facebook Custom Audience (Actions) destination and the Sync Audience action, then click **Create Mapping**. 
4. Enter a descriptive name for your mapping. Segment recommends a name that includes both the audience name and sync mode, for example, `Loyalty Users (Remove)`. 
5. Under **Select record to map and send**, select **Deleted records**. The Deleted records sync mode removes any records from your custom audience that you deleted from your source model.
6. Set how often your model syncs by setting the [Sync schedule](/docs/connections/reverse-etl/#step-4-create-mappings). 
7. Select or create an audience in Facebook to sync your data with. Click the **Select or create audience in Facebook** button to save the audience ID to your mapping.

> 8. Map your model columns to the appropriate Facebook Custom Audience parameters. For more context about data formatting, see the [Sync audience](#sync-audience) and [Data processing](#data-processing) documentation. Map your External ID to a unique user identifier from your system (like User ID, CRM ID, or anonymous ID.) Segment recommends using the External ID column as your primary key when setting up your Reverse ETL model so you can more easily remove users from your custom audience. External ID is the only field Segment requires you to map.

Reviewer: "For more context about data formatting, see the [Sync audience](#sync-audience) and [Data processing](#data-processing) documentation. Map your External ID to a unique user identifier from your system (like User ID, CRM ID, or anonymous ID.) Segment recommends using the External ID column as your primary key when setting up your Reverse ETL model so you can more easily remove users from your custom audience. External ID is the only field Segment requires you to map."

should be:

"Only the External ID is required. When a record is deleted from your source model, only the model primary key is sent to the mapping; other columns from your source model are not sent. We highly recommend using the External ID as your primary key in your source model."

> 8. Map your model columns to the appropriate Facebook Custom Audience parameters. For more context about data formatting, see the [Sync audience](#sync-audience) and [Data processing](#data-processing) documentation. Map your External ID to a unique user identifier from your system (like User ID, CRM ID, or anonymous ID.) Segment recommends using the External ID column as your primary key when setting up your Reverse ETL model so you can more easily remove users from your custom audience. External ID is the only field Segment requires you to map.

Author: ```suggestion
8. Map your model columns to the appropriate Facebook Custom Audience parameters. Only the External ID is required. When a record is deleted from your source model, only the model primary key is sent to the mapping; other columns from your source model are not sent. Segment recommends using the External ID as your primary key in your source model.
```

> 9. Send a test record. If successful, you should see a 200 response in Segment and one record removed from your custom audience. To verify that the record was successfully removed from your custom audience, open Facebook Ads Manager and navigate to **Audiences > {Audience Name} > History**.

Reviewer 2: ```suggestion
9. Send a test record. If successful, you should see a `200` response in Segment and one record removed from your custom audience. To verify that the record was successfully removed from your custom audience, open Facebook Ads Manager and navigate to **Audiences > {Audience Name} > History**.
```

10. Click **Save Mapping** and enable the mapping.

{% include components/actions-fields.html %}

## Data processing

To improve match rates, Segment built in normalization and hashing for common fields to align with [Facebook’s recommended best practices](https://developers.facebook.com/docs/marketing-api/audiences/guides/custom-audiences/#hash){:target="_blank”}.

### Normalization
Segment automatically strips whitespace and converts the following fields to lowercase:
* Email
* First name
* Last name
* First initial
* City
* State
* Country

Segment normalizes the Phone field by removing any non-numeric symbols, whitespace, and leading zeroes.

### Hashing
Facebook requires you to hash all PII before sending it to the Facebook Conversions API. 

> Segment hashes the following fields by default:

Reviewer: Update to:

"Segment will automatically hash any of the following fields that are not already SHA256 hashed at egress:"

> Segment hashes the following fields by default:

Author: ```suggestion
Segment automatically hashes any of the following fields that are not already SHA256 hashed at egress:
```
